src/pages/EventDetails/components/EventRenderer.tsx

```diff
@@ -240,10 +242,49 @@ const EventRenderer = ({ data, preview = false }: EventRendererProps) => {
     );
   };
 
+  type FavoriteProps = IconButtonProps & {
+    eventId: Event['id'];
+  };
+
+  const Favorite = ({ eventId, ...props }: FavoriteProps) => {
+    const { data: favorite } = useEventIsFavorite(eventId);
+    const updateFavorite = useEventSetIsFavorite(eventId);
+    const toggleFavorite = (isFavorite: boolean) => {
+      event('mark-as-favorite', 'event', `Marked event (${data.title}) as favorite`);
```

Comment:
Jeg ser vi sender data til analytics flere steder på samme måte som dette, men da vet vi ikke om handlingen failer eller ikke. Kunne det vært en ide å logge en melding for onSuccess og en for onError? Da ville vi fått mer innsikt når ting går galt.

_fks:_
```
event('mark-as-favorite', 'event', `Marked event (${data.title}, 'error') as favorite`);
```

Reply:
Tja, poenget med analytics føler jeg er å vite hva brukerne gjør, ikke om api'et funker